src/app/public/modules/modal-provider/types/modal-provider.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  EventEmitter
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs';
+
+import {
+  SkyModalProviderCloseArgs
+} from './modal-provider-close-args';
+
+export interface SkyModalProvider {
+
+  type: string;
+  closeCallback: Observable<SkyModalProviderCloseArgs>;
+  events?: { [key: string]: EventEmitter<any> };
```

Comment:
What types of events does this encompass? DOM? Observables?

Reply:
I wrote this JS Doc for this. Let me know if that doesn't make sense.

```
Events which are triggered from inside the modal which are then listened to by the consumer.
```

I also change the type here to `Observable` as that is probably better than `EventEmitter` in this case.